src/components/application_manager/test/commands/mobile/alert_maneuver_request_test.cc

```diff
@@ -0,0 +1,189 @@
+
+  ON_CALL(app_mngr_, application(_))
+      .WillByDefault(Return(ApplicationSharedPtr()));
+
+  MessageSharedPtr result_msg(CatchMobileCommandResult(CallRun(*command)));
+  EXPECT_EQ(mobile_apis::Result::APPLICATION_NOT_REGISTERED,
+            static_cast<mobile_apis::Result::eType>(
+                (*result_msg)[am::strings::msg_params][am::strings::result_code]
+                    .asInt()));
+}
+
+TEST_F(AlertManeuverRequestTest, Run_ProcessingResult_UNSUCCESS) {
+  MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+  (*msg)[am::strings::msg_params][am::strings::soft_buttons] = 0;
+
+  CommandPtr command(CreateCommand<AlertManeuverRequest>(msg));
+
+  MockAppPtr app(CreateMockApp());
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(app));
+
+  ON_CALL(app_mngr_, GetPolicyHandler())
+      .WillByDefault(
+          ReturnRef(*static_cast<policy::PolicyHandlerInterface*>(NULL)));
+
+  const mobile_apis::Result::eType kProcessingResult =
+      mobile_apis::Result::ABORTED;
+
+  EXPECT_CALL(*(am::MockMessageHelper::message_helper_mock()),
+              ProcessSoftButtons(_, _, _, _))
+      .WillOnce(Return(kProcessingResult));
+
+  MessageSharedPtr result_msg(CatchMobileCommandResult(CallRun(*command)));
+  EXPECT_EQ(kProcessingResult,
+            static_cast<mobile_apis::Result::eType>(
+                (*result_msg)[am::strings::msg_params][am::strings::result_code]
+                    .asInt()));
+}
+
+TEST_F(AlertManeuverRequestTest, Run_IsWhiteSpaceExist_UNSUCCESS) {
+  MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+  (*msg)[am::strings::msg_params][am::strings::soft_buttons] = 0;
+  (*msg)[am::strings::msg_params][am::strings::tts_chunks] =
+      SmartObject(smart_objects::SmartType_Array);
+
+  SmartObject tts_chunk(smart_objects::SmartType_Map);
+  tts_chunk[am::strings::text] = "wrong chunk syntax\t\n";
+
+  (*msg)[am::strings::msg_params][am::strings::tts_chunks].asArray()->push_back(
+      tts_chunk);
+
+  CommandPtr command(CreateCommand<AlertManeuverRequest>(msg));
+
+  MockAppPtr app(CreateMockApp());
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(app));
+
+  MessageSharedPtr result_msg(CatchMobileCommandResult(CallRun(*command)));
+  EXPECT_EQ(mobile_apis::Result::INVALID_DATA,
+            static_cast<mobile_apis::Result::eType>(
+                (*result_msg)[am::strings::msg_params][am::strings::result_code]
+                    .asInt()));
+}
+
+TEST_F(AlertManeuverRequestTest, Run_ProcessingResult_SUCCESS) {
+  MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+  (*msg)[am::strings::msg_params][am::strings::soft_buttons] = 0;
+
+  CommandPtr command(CreateCommand<AlertManeuverRequest>(msg));
+
+  MockAppPtr app(CreateMockApp());
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(app));
+
+  ON_CALL(app_mngr_, GetPolicyHandler())
+      .WillByDefault(
+          ReturnRef(*static_cast<policy::PolicyHandlerInterface*>(NULL)));
```

Comment:
@vlantonov , I didn't checked it. Just moved it from develop to release/4.2.0 branch.
It should be created another task in JIRA and assigned to PM for fixing all issues in source code. 
Because I have a time limit for this task, which is supposing only moving unit tests from branch to branch.
Fixed a2da495